tools/regs2groov.py

```diff
@@ -0,0 +1,134 @@
+import xml.etree.ElementTree as et
+import re
+from collections import namedtuple
+import argparse
+
+
+Group = namedtuple('Group', ['name', 'registers'])
+Register = namedtuple('Register', ['name', 'address', 'fields'])
+Field = namedtuple('Field', ['name', 'access', 'msb', 'lsb'])
+
+def parse_svd(filename):
+    def mk_field(x):
+        (msb, lsb) = re.search(r"\[(\d+):(\d+)\]", x.find('bitRange').text).groups()
+
+        return Field(
+            name = x.find('name').text,
+            access = x.find('access').text,
+            msb = int(msb),
+            lsb = int(lsb)
+        )
+
+    def mk_register(x, base):
+        return Register(
+            name = x.find('name').text,
+            address = base + int(x.find('addressOffset').text[2:], 16),
+            fields = [mk_field(f) for f in x.findall('.//field')]
+        )
+
+    def mk_group(x, root):
+        base_addr = int(x.find('baseAddress').text[2:], 16)
```

Comment:
**note**: If the `[2:]` here is skipping an `0x` prefix, there's no need - parsing an int from base 16 already handles that.

Reply:
Nice 👍🏻